File: Leetcode/Leetcode.2009.minimum-number-of-operations-to-make-array-continuous/Leetcode.2009.minimum-number-of-operations-to-make-array-continuous.cpp
class Solution { // Hinted by (sort this array)
public:

    int minOperations(vector<int>& nums_v) {
        const size_t size(nums_v.size());
        int* nums(nums_v.data());
        sort(nums + 0, nums + size);
        // for (size_t i(0); i < size; ++i) {
        //     cout << nums[i] << ',';
        // }
        size_t i(0);
        size_t j(0);

        size_t length(size - 1);

        // set<int> s;
        // s.insert(nums[0]);
        
        size_t count(0);
        size_t max(0);
        while (j < size) {
            int to_add(nums[j]);
            ++count;
            ++j;
            while (j < size && nums[j] == to_add) {
                ++j;
            }
            
            size_t curr_length(nums[j - 1] - nums[i]);
            while (i < j && curr_length > length) {
                int to_remove(nums[i]);
                --count;
                ++i;
                while (i < j && nums[i] == to_remove) {
                    ++i;
                }
                curr_length = nums[j - 1] - nums[i];
            }
            // cout << i << ' ' << j << ' ' << count << '\n';
            if (count > max) {
                max = count;
            }
        }
        // cout << max;
        return (int)size - (int)max;
    }
};

File: Leetcode/Leetcode.2009.minimum-number-of-operations-to-make-array-continuous/Leetcode.2009.minimum-number-of-operations-to-make-array-continuous.rs
impl Solution { // Hinted by (sort this array)
    pub fn min_operations(nums0: Vec<i32>) -> i32 {
        let mut nums: Vec<i32> = nums0;
        let len: usize = nums.len();
        let target: usize = len - 1;
        nums.sort();
        let mut i: usize = 0;
        let mut j: usize = 0;

        let mut count: usize = 0;
        let mut max: usize = 0;

        while j < len {
            let to_add: i32 = nums[j];
            count += 1;
            j += 1;
            while j < len && nums[j] == to_add {
                j += 1;
            }
            let mut curr_length: usize = (nums[j - 1] - nums[i]) as usize;
            while i < j && curr_length > target {
                let to_rm: i32 = nums[i];
                i += 1;
                count -= 1;
                while i < j && nums[i] == to_rm {
                    i += 1;
                }
                curr_length = (nums[j - 1] - nums[i]) as usize;
            }
            if count > max {
                max = count;
            }
        }
        (len - max) as i32
    }
}